core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -661,43 +665,37 @@ private void terminate(String cacheName) {
    @Override
    public void stop() {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
-      if (!stopping) {
-         synchronized (this) {
-            // DCL to make sure that only one thread calls stop at one time,
-            // and any other calls by other threads are ignored.
-            if (!stopping) {
-               log.debugf("Stopping cache manager %s on %s", configurationManager.getGlobalConfiguration().transport().clusterName(), getAddress());
-               stopping = true;
-               Set<String> cachesToStop = new LinkedHashSet<>(this.caches.size());
-               boolean defaultCacheHasDependency = false;
-               // stop ordered caches first
-               try {
-                  List<String> ordered = cacheDependencyGraph.topologicalSort();
-                  defaultCacheHasDependency = ordered.contains(DEFAULT_CACHE_NAME);
-                  cachesToStop.addAll(ordered);
-               } catch (CyclicDependencyException e) {
-                  log.stopOrderIgnored();
-               }
-               cachesToStop.addAll(caches.keySet());
-               // will be stopped by the GCR
-               InternalCacheRegistry internalCacheRegistry = globalComponentRegistry.getComponent(InternalCacheRegistry.class);
-               internalCacheRegistry.filterPrivateCaches(cachesToStop);
-               // make sure we stop the default cache LAST!
-               if(!defaultCacheHasDependency) {
-                  cachesToStop.add(DEFAULT_CACHE_NAME);
-               }
-               for (String cacheName : cachesToStop) {
-                  terminate(cacheName);
-               }
-               globalComponentRegistry.getComponent(CacheManagerJmxRegistration.class).stop();
-               globalComponentRegistry.stop();
 
-            } else {
-               log.trace("Ignore call to stop as the cache manager is stopping");
-            }
+      synchronized (this) {
+         if (stopping) {
+            log.trace("Ignore call to stop as the cache manager is stopping");
+            return;
```

Comment:
seems we can remove the return and just have the rest of the code in an else block